javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserClassDeclaration.java

```diff
@@ -296,10 +306,11 @@ public boolean isTypeParameter() {
         List<ResolvedReferenceType> ancestors = new ArrayList<>();
 
         // We want to avoid infinite recursion in case of Object having Object as ancestor
-        if (!(Object.class.getCanonicalName().equals(getQualifiedName()))) {
+        if (!this.isJavaLangObject()) {
```

Comment:
Just spotted this -- I intend to remove this, in favour of this (and then the content can be un-indented)...

```java
if(this.isJavaLangObject()) {
    return ancestors; // empty list
}
```

Reply:
Done